internal/harnesses/remote/k8s.go

```diff
@@ -0,0 +1,382 @@
+			return err
+		}
+
+		if ready {
+			break
+		}
+	}
+
+	return nil
+}
+
+func (h *k8s) createRbacResources(ctx context.Context, namespaceName string, commonLabels map[string]string) error {
+	rbacResourcesName := fmt.Sprintf(RbacResourcesNameFormat, h.Id)
+
+	_, err := h.Client.RbacV1().ClusterRoles().Create(ctx, &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   rbacResourcesName,
+			Labels: commonLabels,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:     []string{"get", "create", "list", "delete", "watch"},
+				APIGroups: []string{""},
+				Resources: []string{"pods", "configmaps", "secrets", "serviceaccounts", "services", "namespaces"},
+			},
+			{
+				Verbs:     []string{"get", "create", "list", "delete", "watch"},
+				APIGroups: []string{"apps"},
+				Resources: []string{"deployments", "daemonsets", "replicasets", "statefulsets"},
+			},
+			{
+				Verbs:     []string{"get", "create", "list", "delete", "watch"},
+				APIGroups: []string{"batch"},
+				Resources: []string{"jobs", "cronjobs"},
+			},
+			{
+				Verbs:     []string{"get", "create", "list", "delete", "watch"},
+				APIGroups: []string{"storage"},
+				Resources: []string{"jobs", "cronjobs"},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, err = h.Client.CoreV1().ServiceAccounts(namespaceName).Create(ctx, &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rbacResourcesName,
+			Namespace: namespaceName,
+			Labels:    commonLabels,
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, err = h.Client.RbacV1().ClusterRoleBindings().Create(ctx, &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   rbacResourcesName,
+			Labels: commonLabels,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     rbacResourcesName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      rbacResourcesName,
+				Namespace: namespaceName,
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *k8s) Destroy(ctx context.Context) error {
+	sandboxPodName := fmt.Sprintf(ImagetestSandboxPodNameFormat, h.Id)
+	namespaceName := fmt.Sprintf(ImagetestNamespaceNameFormat, h.Id)
+	rbacResourcesName := fmt.Sprintf(RbacResourcesNameFormat, h.Id)
+
+	var gracePeriodSeconds int64 = 0
+
+	err := h.Client.RbacV1().ClusterRoleBindings().Delete(ctx, rbacResourcesName, metav1.DeleteOptions{
+		GracePeriodSeconds: &gracePeriodSeconds,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to delete RoleBinding %q: %w", rbacResourcesName, err)
+	}
+
+	gracePeriodSeconds = 0
+	err = h.Client.RbacV1().ClusterRoles().Delete(ctx, rbacResourcesName, metav1.DeleteOptions{
+		GracePeriodSeconds: &gracePeriodSeconds,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to delete Role %q: %w", rbacResourcesName, err)
+	}
+
+	gracePeriodSeconds = 0
```

Comment:
```suggestion
```

I don't think you need to redefine these?

Reply:
Fair, I was redefining because they're pointer types and that might mean they get updated in the context of the function call.